packages/block-editor/src/components/inner-blocks/index.js

```diff
@@ -160,6 +160,20 @@ export function useInnerBlocksProps( props = {}, options = {} ) {
 			? ControlledInnerBlocks
 			: UncontrolledInnerBlocks;
 
+	const children = [
```

Comment:
@ellatrix What do you think of `childrenBefore`/`childrenAfter`? This is trying to address the situation in which a block that uses the hook has additional inner content it wants to add that is not a block (eg: cite, in this case).

Reply:
It seems pretty weird to me. Why not just do something like:

```jsx
<BlockQuotation { ...blockProps }>
	<div { ...innerBlocksProps } />
	<footer><RichText tagName="cite" /></footer>
</BlockQuotation>

```

This would have the added benefit of having a container to target for styling the content of the quote without including the citation.

Reply:
I also like @ZebulanStanphill's proposal, mixing inner blocks with random markup is something that seems fragile.

Reply:
> This would have the added benefit of having a container to target for styling the content of the quote without including the citation.

One of the reasons this PR exists is that we want to remove the extra nodes in the editor, so it looks like the frontend. However! This gave me an idea to try. What if we did the same but _marking the position_ of the inner blocks somehow without adding a wrapper:

```js
<BlockQuotation { ...blockProps }>
	<Fragment { ...innerBlocksProps } />
	<cite>Cite goes here</cite>
</BlockQuotation>
```

It checkes the boxes (no wrapper, inner blocks live alongside other inner content of the block container, uses a native solution to position everything, etc). @ellatrix Does this look like something we want to promote for this use case?

Reply:
@nosolosw I'm pretty sure that using `innerBlocksProps` on a `Fragment` is **not valid**. There's nowhere to put any CSS classes... or any other HTML attributes for that matter. According to the [React docs](https://reactjs.org/docs/fragments.html#keyed-fragments):

>`key` is the only attribute that can be passed to `Fragment`. In the future, we may add support for additional attributes, such as event handlers.

Reply:
Oh, you're right! Thanks for pointing it out.

So we're back to the drawing board. These are the options I see forward:

1. We make cite part of the inner blocks => this means that we can't control exactly its semantics (position and that there's only one) so some of the [interactions suggested](https://github.com/WordPress/gutenberg/issues/15486#issuecomment-706416700) may be more difficult to implement.
2. We use a wrapper. This seems precisely what we wanted to avoid by using the `useInnerBlocksProps` API to make the editor look like the front-end.
3. Make `useInnerBlocksProp` process existing children somehow. I gave this a try with the `beforeChildren`/`afterChildren` exploration (it worked and checked all the boxes but it was arguably less declarative, so people wasn't a fan). There may be other options to explore.

Thoughts? @ellatrix @youknowriad

Reply:
What's wrong with just having a wrapper element for the quote content (I mean both on the front-end and back-end)? Arguably, a wrapper there would be really handy for themes trying to apply styles to the quote content without affecting the citation.

Reply:
I know little of how the current blocks' markup came to be and some of the InnerBlocks restrictions, so talked to some folks to educate myself about this. This is what I gathered:

- The block editor is good at making the post_content output semantic HTML, without unnecessary wrapper nodes. That's something we need to keep doing.
- InnerBlocks behavior needs a wrapper, can't live alongside other markup within the same root node. I've taken a look at other blocks and quote/pullquote seem to be the only ones affected by this.

These two statements are both true and at odds in the specific use case of quote/pullquote --- other blocks such as gallery (images + caption) already wrap the images within a UL, so it can be used as the innerBlocks wrapper.

I'm going to leave this a few days on the backburner and will come back if there're new alternatives to try or folks show a strong preference for one of the above.

cc @youknowriad @mtias @ellatrix

Reply:
@nosolosw @youknowriad @ZebulanStanphill The inner blocks hook was designed to handle this. I took the liberty to push a fix in 252be93844ce6f9bb44c9b8d4aed570c1339df4e. Let me know if there's anything that doesn't work as expected.

I even used the quote block as an example in the API proposal. https://github.com/WordPress/gutenberg/pull/25633

Reply:
Thanks, Ella, this is nice! I think one of the last concerns people shared with me was this one: in the future, inner blocks may need a wrapper to allow things like positioning elements within using flex/grid/etc.

My understanding is that, if we need that, it's going to affect both the editor & the front-end markup, hence, we'd need to change the blockquote output, no matter whether the block uses the current approach in master or this one. If anything, it looks like this PR would make it easier to do that change.

So, if there're no concerns against this, I'm taking it to the finish line (there're some tests that need fixing, etc).

Reply:
> I think one of the last concerns people shared with me was this one: in the future, inner blocks may need a wrapper to allow things like positioning elements within using flex/grid/etc.

I'm not sure I understand why would this be necessary? Could you give an example?

Reply:
Not sure I'm following either. It's always possible to use a dedicated wrapper for inner block or not. Up to the implementation. Inner blocks doesn't have restrictions to what can be in the wrapper element, InnerBlock.Content was like this from the start.